specification/databricks/resource-manager/Microsoft.Databricks/stable/2024-05-01/databricks.json

```diff
@@ -0,0 +1,1991 @@
+        "enhancedSecurityMonitoring": {
+          "type": "object",
+          "description": "Status of Enhanced Security Monitoring feature.",
+          "$ref": "#/definitions/EnhancedSecurityMonitoringDefinition"
+        }
+      }
+    },
+    "AutomaticClusterUpdateDefinition": {
+      "type": "object",
+      "description": "Status of automated cluster updates feature.",
+      "properties": {
+        "value": {
+          "type": "string",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "AutomaticClusterUpdateValue"
+          }
+        }
+      }
+    },
+    "EnhancedSecurityMonitoringDefinition": {
+      "type": "object",
+      "description": "Status of Enhanced Security Monitoring feature.",
+      "properties": {
+        "value": {
+          "type": "string",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "EnhancedSecurityMonitoringValue"
+          }
+        }
+      }
+    },
+    "ComplianceSecurityProfileDefinition": {
+      "type": "object",
+      "description": "Status of Compliance Security Profile feature.",
+      "properties": {
+        "complianceStandards": {
+          "description": "Compliance standards associated with the workspace.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ComplianceStandardDefinition"
+          }
+        },
+        "value": {
+          "type": "string",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "ComplianceSecurityProfileValue"
+          }
+        }
+      }
+    },
+    "ComplianceStandardDefinition": {
+      "description": "Compliance standard that can be associated with a workspace.",
+      "type": "string",
+      "enum": [
+        "NONE",
+        "HIPAA",
+        "PCI_DSS"
+      ],
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "ComplianceStandard"
+      }
+    },
+    "WorkspaceCustomParameterType": {
+      "description": "Provisioning status of the workspace.",
+      "readOnly": true,
+      "enum": [
+        "Bool",
+        "Object",
+        "String"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "CustomParameterType",
+        "modelAsString": true
+      }
+    },
+    "WorkspaceNoPublicIPBooleanParameter": {
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/WorkspaceCustomParameterType",
+          "description": "The type of variable that this is"
+        },
+        "value": {
+          "type": "boolean",
+          "description": "The value which should be used for this field.",
+          "default": true
+        }
+      },
+      "type": "object",
+      "required": [
```

Comment:
@mvvsubbu Isn't WorkspaceNoPublicIPBooleanParameter.value marked as required here?

Reply:
@TimLovellSmith like we have discussed before properties.parameters.WorkspaceNoPublicIPBooleanParameter is not required, but when the customer decides to provide properties.parameters.WorkspaceNoPublicIPBooleanParameter then we ask him to provider properties.parameters.WorkspaceNoPublicIPBooleanParameter.value as well. I hope that helps why it is not required.

Reply:
Let me know if we still have questions, we can hop on a call and discuss.

Reply:
@mvvsubbu I think @TimLovellSmith is referring to line 1190 where it is marked 
      "required": [
        "value"
      ],
Previously NoPublicIPBooleanParameter was pointing to WorkspaceCustomBooleanParameter which also had a required parameter so the WorkspaceNoPublicIPBooleanParameter was created to follow the same convention

Reply:
@curiekim0 I had offline chat with @mikekistler he suggested we should remove "required" attribute for WorkspaceNoPublicIPBooleanParameter. Can you do this change, let me know if you have any questions. I don't want us to do more changes and get into more errors again :D

Reply:
@mikekistler made changes as per our offline discussion, please approve the PR